sdk/core/azure-core/azure/core/exceptions.py

```diff
@@ -333,6 +333,16 @@ def _parse_odata_body(error_format, response):
             pass
         return None
 
+    def __str__(self):
+        retval = super(HttpResponseError, self).__str__()
```

Comment:
Currently, str(HttpResponseError) includes message information. I don't think we want to lose it?

Reply:
i'm not losing any message information, i'm just adding on the response body if it exists to the end of `str(HtpResponseError)`

Reply:
You can extend the test https://github.com/Azure/azure-sdk-for-python/blob/main/sdk/core/azure-core/tests/test_exceptions.py#L128 and make sure we don't dup the error message for ODataV4 format